lab_5_scrapper/scrapper.py

```diff
@@ -1,82 +1,149 @@
 """
 Crawler implementation
 """
+import datetime
+import json
+from pathlib import Path
 from typing import Pattern, Union
+from urllib.parse import urlparse
 
+import requests
+from bs4 import BeautifulSoup
+
+from core_utils.article.article import Article
+from core_utils.article.io import to_meta, to_raw
+from core_utils.config_dto import ConfigDTO
+from core_utils.constants import (ASSETS_PATH, CRAWLER_CONFIG_PATH,
+                                  NUM_ARTICLES_UPPER_LIMIT,
+                                  TIMEOUT_LOWER_LIMIT, TIMEOUT_UPPER_LIMIT)
+
+class IncorrectSeedURLError(Exception):
+pass
+
+
+class NumberOfArticlesOutOfRangeError(Exception):
+pass
+
+class IncorrectNumberOfArticlesError(Exception):
+pass
+
+class IncorrectHeadersError(Exception):
+pass
+
+class IncorrectEncodingError(Exception):
+pass
+
+class IncorrectTimeoutError(Exception):
+pass
+
+class IncorrectVerifyError(Exception):
+pass
 
 class Config:
     """
     Unpacks and validates configurations
     """
+    seed_urls: list[str]
+    total_articles_to_find_and_parse: int
+    headers: dict[str, str]
+    encoding: str
+    timeout: int
+    verify_certificate: bool
+    headless_mode: bool
 
     def __init__(self, path_to_config: Path) -> None:
         """
         Initializes an instance of the Config class
         """
-        pass
+        self.path_to_config = path_to_config
+        self._validate_config_content()
+        configDTO = self._extract_config_content()
+        self._seed_urls = configDTO.seed_urls
+        self._num_articles = configDTO.total_articles
+        self._headers = configDTO.headers
+        self._encoding = configDTO.encoding
+        self._timeout = configDTO.timeout
+        self._should_verify_certificate = configDTO.should_verify_certificate
+        self._headless_mode = configDTO.headless_mode
 
     def _extract_config_content(self) -> ConfigDTO:
         """
         Returns config values
         """
-        pass
+        with open (self.path_to_config, 'r', encoding='utf-8') as f:
+            config = json.load(f)
+        return ConfigDTO(**config)
+
 
     def _validate_config_content(self) -> None:
         """
         Ensure configuration parameters
         are not corrupt
         """
+        with open(self.path_to_config, 'r', encoding='utf-8') as f:
+            config = json.load(f)
+        seed_urls = config['seed_urls']
+        headers = config['headers']
+        total_articles_to_find_and_parse = config['total_articles_to_find_and_parse']
+        encoding = config['encoding']
+        timeout = config['timeout']
+        verify_certificate = config['should_verify_certificate']
+        headless_mode = config['headless_mode']
+
         pass
 
     def get_seed_urls(self) -> list[str]:
         """
         Retrieve seed urls
         """
-        pass
+        return configDTO.seed_urls
```

Comment:
you have specific class attributes for all getting methods -> there is no need to get them from configDTO -> get them from class fields directly e.g. here it is _seed_urls